tfe/resource_tfe_team.go

```diff
@@ -24,26 +27,60 @@ func resourceTFETeam() *schema.Resource {
 				Required: true,
 				ForceNew: true,
 			},
-
 			"organization": {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
 			},
+			"organization_access": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"manage_policies": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"manage_workspaces": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"manage_vcs_settings": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"visibility": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```

Comment:
Hmm I'm not sure we want this to be marked as a computed value. `Computed` is used to represent values that are not user configurable or aren't known at time of an `apply` or `plan` operation, which this shouldn't be.

Reply:
I set this based on the following comment from the plugin SDK:

```
// If Computed is true, then the result of this value is computed
// (unless specified by config) on creation.
```

I take that to mean that any with a server set default could be "computed," though this probably moot when it's an optional field where the zero value of the type is equal to the server value. If the server were capable of defaulting to `true`, I think this might need to be set though? I can remove it here.

Reply:
Hmm, was mis-remembering what values `visibility` could have. I think this needs to be computed since the API will respond with `visibility = "secret"` regardless of whether the user has set a value and we'll persist that value to state. I would expect Terraform to present a diff from `"secret" -> null` on subsequent runs if we don't account for this. I think computed takes care of this, but correct me if I'm wrong.

Reply:
> since the API will respond with visibility = "secret" regardless of whether the user has set a value 

Nope, the API should respond with whatever value the user set - and valid options are `"secret"` or `"organization"`. The default is `"secret"`.

* `Computed` is very much for unknown values at plan-time that are _not_ user-configurable - things like dates and UUIDs.
* `Default` is for known values to be used when something isn't explicitly set within the configuration.

For our case here, these are all values that have known defaults that we can provide at plan-time; so they can all be defaulted appropriately. (So this visibility section should not be computed, and should have a default of `"secret"`).

When that's all finished, given a configuration like this:

```hcl
resource "tfe_team" "test-1" {
  name         = "my-team-name"
  organization = var.organization
  organization_access {
    manage_policies = true
    manage_workspaces = false
  }
}
```

We should see a plan output like this:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # tfe_team.test-1 will be created
  + resource "tfe_team" "test-1" {
      + id           = (known after apply)
      + name         = "my-team-name"
      + organization = "my-cool-organization"
      + visibility   = "secret"

      + organization_access {
          + manage_policies     = true
          + manage_vcs_settings = false
          + manage_workspaces   = false
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

See the [Schema Behaviors documentation](https://www.terraform.io/docs/extend/schemas/schema-behaviors.html) for more info on both of these behaviors.